src/parcsr_mv/par_csr_matvec.c

```diff
@@ -268,7 +272,15 @@ hypre_ParCSRMatrixMatvecOutOfPlace( HYPRE_Complex       alpha,
    }
 
 #ifdef HYPRE_PROFILE
-   hypre_profile_times[HYPRE_TIMER_ID_PACK_UNPACK]   += hypre_MPI_Wtime();
+   hypre_profile_times[HYPRE_TIMER_ID_PACK_UNPACK] += hypre_MPI_Wtime();
+#endif
+
+   /* when using GPUs, start local matvec first in order to overlap with communication */
+#if defined(HYPRE_USING_GPU)
+   hypre_CSRMatrixMatvecOutOfPlace( alpha, diag, x_local, beta, b_local, y_local, 0 );
+#endif
+
+#ifdef HYPRE_PROFILE
    hypre_profile_times[HYPRE_TIMER_ID_HALO_EXCHANGE] -= hypre_MPI_Wtime();
 #endif
 
```

Comment:
@pwang234 I don't recall why we did not need a `hypre_ForceSyncComputeStream` for the next `hypre_ParCSRCommHandleCreate_v2`. Is it because the above `thrust::gather` does a sync? If so, it is a dangerous assumption here. Thanks!